data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/handlers/beacon/GetCommitteesTest.java

```diff
@@ -101,20 +86,21 @@ public void shouldReturnBadRequestWhenNoEpochIsSupplied() throws Exception {
   }
 
   @Test
-  public void shouldReturnEmptyListWhenAFutureEpochIsRequested() throws Exception {
+  public void shouldHandleMissingResult() throws Exception {
     final GetCommittees handler = new GetCommittees(provider, jsonProvider);
-    final UnsignedLong futureEpoch = slot.plus(UnsignedLong.valueOf(SLOTS_PER_EPOCH));
 
+    when(context.queryParamMap()).thenReturn(Map.of(EPOCH, List.of("0")));
     when(provider.isStoreAvailable()).thenReturn(true);
-    when(provider.getCommitteesAtEpoch(futureEpoch))
-        .thenReturn(SafeFuture.completedFuture(List.of()));
-    when(context.queryParamMap()).thenReturn(Map.of(EPOCH, List.of(futureEpoch.toString())));
+    when(provider.getCommitteesAtEpoch(ZERO))
+        .thenReturn(SafeFuture.completedFuture(Optional.empty()));
     handler.handle(context);
 
     verify(context).result(args.capture());
     verify(context).header(Header.CACHE_CONTROL, CACHE_NONE);
```

Comment:
if we're returning a result of 'SC_GONE', that can totally be cached, so this does look like a scenario we want to dig into if cache says none here. We'd be better off not setting the header at all and letting default behaviour happen potentially

Reply:
Added tests to check the cache.  If we're querying against a finalized state, we'll set `CACHE_FINALIZED` otherwise `CACHE_NONE`.